pkg/operator/cluster.go

```diff
@@ -33,42 +33,58 @@ import (
 	kwatch "k8s.io/client-go/pkg/watch"
 )
 
-type clusterTPR struct {
+type clusterManager struct {
 	context      *context
+	name         string
 	watchVersion string
 	devicesInUse bool
 	clusters     map[string]*cluster.Cluster
 	tracker      *tprTracker
 	sync.RWMutex
+	// The initiators that create TPRs specific to specific Rook clusters.
+	// For example, pools, object services, and file services, only make sense in the context of a Rook cluster
+	inclusterInitiators []inclusterInitiator
+	inclusterMgrs       []tprManager
 }
 
-func newClusterTPR(context *context) *clusterTPR {
-	return &clusterTPR{
-		context:  context,
-		clusters: make(map[string]*cluster.Cluster),
-		tracker:  newTPRTracker(),
+func newClusterManager(context *context, inclusterInitiators []inclusterInitiator) *clusterManager {
+	return &clusterManager{
+		context:             context,
+		clusters:            make(map[string]*cluster.Cluster),
+		tracker:             newTPRTracker(),
+		inclusterInitiators: inclusterInitiators,
 	}
 }
 
-func (t *clusterTPR) Name() string {
+// Gets the name of the TPR
+func (t *clusterManager) Name() string {
 	return "rookcluster"
 }
 
-func (t *clusterTPR) Description() string {
+// Gets the description of the TPR
+func (t *clusterManager) Description() string {
 	return "Managed Rook clusters"
 }
 
-func (t *clusterTPR) Load() error {
+func (t *clusterManager) Manage() {
+	for {
+		logger.Infof("Managing clusters")
+		err := t.Load()
+		if err != nil {
+			logger.Errorf("failed to load cluster. %+v", err)
+		} else {
+			if err := t.Watch(); err != nil {
+				logger.Errorf("failed to watch clusters. %+v", err)
+			}
+		}
 
-	// Check if there is an existing cluster to recover
-	err := t.findAllClusters()
-	if err != nil {
-		return fmt.Errorf("failed to find clusters. %+v", err)
+		<-time.After(time.Second * time.Duration(t.context.retryDelay))
 	}
-	return nil
 }
 
-func (t *clusterTPR) findAllClusters() error {
+func (t *clusterManager) Load() error {
```

Comment:
using `t` as the function receiver name is a bit odd now that the type has changed